enumerable_methods.rb

```diff
@@ -0,0 +1,132 @@
+module Enumerable
+  # start of my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+  # end of my_each
+
+  # start of my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+  # end of my_each_with_index
+
+  # my_select
+  def my_select
```

Comment:
- [ ] [`select`](https://apidock.com/ruby/Array/select) returns a new array containing all elements of array for which the given block returns true value.

Your current implementation doesn't return the `new_array`, it returns nil.

Also in `new_array << i` you are appending the index, not the element itself.